### PR TITLE
ENH: Add lazy copy for sort_index

### PR DESCRIPTION
**Pull request by phofl**
- [ ] xref #49473 (Replace xxxx with the GitHub issue number)
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

File: pandas/conftest.py [hunk 1]
```diff
@@ -1903,6 +1903,7 @@ def using_copy_on_write() -> bool:
     """
     Fixture to check if Copy-on-Write is enabled.
     """
+    pd.options.mode.copy_on_write = True
```
Comment (jorisvandenbossche) [inline pandas/conftest.py:1906]: I suppose this is a leftover from local testing?

**Review by jorisvandenbossche — COMMENTED** [review]
Looks good! Just one left-over comment

File: pandas/conftest.py (hunk 1 above)
Comment (phofl) [reply pandas/conftest.py]: Yes, thanks for noticing

Comment (jorisvandenbossche) [reply pandas/conftest.py]: BTW, for testing, you can set the env variable `PANDAS_COPY_ON_WRITE=1 pytest ...`, which is of course a bit verbose. What I do locally is that I have an alias in my bashrc, so I can just do `COW pytest ...`, and then it is quite easy to switch between `COW pytest ...` or `pytest ...` when I want to run a specific test while developing with both COW enabled or not.

Comment (phofl) [reply pandas/conftest.py]: Yeah agreed, did this when running tests in PyCharm

Comment (jorisvandenbossche) [reply pandas/conftest.py]: Ah, I can't speak for PyCharm ;)

**Pull request merged**